mathesar/api/pagination.py

```diff
@@ -78,7 +78,7 @@ def paginate_queryset(
         preview_metadata = None
         # Only tables have columns on the Service layer that hold data necessary for preview template.
         if isinstance(table, Table):
-            columns_query = table.columns.all()
+            columns_query = Column.objects.filter(table_id=table.id).select_related('table__schema__database').prefetch('name')
```

Comment:
How unfortunate that `table.columns` doesn't work here.

Reply:
I was annoyed by that too and tried looking for a better solution, but the [Django suggested solution](https://docs.djangoproject.com/en/1.10/releases/1.10/#manager-use-for-related-fields-and-inheritance-changes) doesn't seem to be working, I will make this refactor sometime later